### PR TITLE
Added public endpoint for fetching subset of projects info

### PR DESCRIPTION
**Pull request by this-Aditya**
Added a new endpoint for fetching the project data to be used for the self-enrollment portal

Fixes https://github.com/RADAR-base/ManagementPortal/issues/915

### CODE REVIEW

**Comment by Bdegraaf1234** [comment]
Nice work @this-Aditya,

Some thoughts, feel free to override:

- We already have a class MinimumprojectDetails, is that perhaps suitable to expand here?
- We might benefit from naming these public endpoints something like `api/projects/public` and putting it in the `ProjectResource`.
- Alternatively, I would switch to projects-lite so that any alphabetically ordered endpoint listing will group this with `api/projects`

I guess I just logically group this with Project a little more than is done in the current implementation. If others disagree (@yatharthranjan @mpgxvii) feel free to approve and move forward

**Comment by this-Aditya** [comment]
Thanks for the review, @Bdegraaf1234 !
I’ll make the desired modifications. Additionally, would it be acceptable to also retrieve the updated `MinimalProjectDetails` when fetching from the `api/projects` endpoint with the `minimized` parameter set to `true`?

**Comment by Bdegraaf1234** [comment]
Yeah that might be an issue, I think it would be good to check with if someone knows where we use the `minimized` endpoint..

Are you in the SEP meeting later today? Would be good to discuss this there

**Comment by this-Aditya** [comment]
Okay, fine. I don't have an invite for that meeting. Could you please share the link on Slack? 
Thanks.

File: src/main/java/org/radarbase/management/service/ProjectService.kt [hunk 1]
```diff
@@ -69,6 +69,17 @@ class ProjectService(
         }
     }
 
+    /**
+     *Get all the projects for the public endpoint.
+     *
+     *@return the list of ProjectLiteDTO
+     */
+    @Transactional(readOnly = true)
+    fun findProjectsForPublicEndpoint(pageable: Pageable): Page<*> {
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/service/ProjectService.kt:78]: `getPublicProjects`

File: src/main/java/org/radarbase/management/config/OAuth2ServerConfiguration.kt [hunk 2]
```diff
@@ -129,6 +129,7 @@ class OAuth2ServerConfiguration(
                 .skipUrlPattern(HttpMethod.GET, "/management/health")
                 .skipUrlPattern(HttpMethod.POST, "/oauth/token")
                 .skipUrlPattern(HttpMethod.GET, "/api/meta-token/*")
+                .skipUrlPattern(HttpMethod.GET, "/api/lite-projects")
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/config/OAuth2ServerConfiguration.kt:132]: I agree with Bastiaan, this should be something like /api/public/projects

File: src/main/java/org/radarbase/management/service/dto/ProjectLiteDTO.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    var projectName: String? = null
+
+    @NonNull
+    var description: String? = null
+
+    @NotNull
+    var location: String? = null
+    var projectStatus: ProjectStatus? = null
+
+
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/service/dto/ProjectLiteDTO.kt:21]: we need source-types here

File: src/main/java/org/radarbase/management/service/dto/ProjectLiteDTO.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package org.radarbase.management.service.dto
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import org.radarbase.management.domain.enumeration.ProjectStatus
+import reactor.util.annotation.NonNull
+import java.io.Serializable
+import javax.validation.constraints.NotNull
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+class ProjectLiteDTO : Serializable {
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/service/dto/ProjectLiteDTO.kt:10]: `PublicProjectDTO`

File: src/main/java/org/radarbase/management/web/rest/ProjectLiteResource.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * GET  /lite-projects : get all the projects.
+ *
+ * @return the ResponseEntity with status 200 (OK) and the list of ProjectLiteDTO
+ */
+
+@RestController
+@RequestMapping("/api")
+class ProjectLiteResource(
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/web/rest/ProjectLiteResource.kt:24]: You can call this `PublicResource` with annotation `@RequestMapping("/api/public")` and then we can house all the public endpoints in the future here.

File: src/main/java/org/radarbase/management/web/rest/ProjectLiteResource.kt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @Autowired private val projectService: ProjectService
+) {
+
+    @GetMapping("lite-projects")
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/web/rest/ProjectLiteResource.kt:28]: should be just `projects` with the above path.

**Review by yatharthranjan — COMMENTED** [review]
I agree with most of Bastiaan's suggestions. Adding a new DTO instead of extending the previous one is acceptable, as this has a specific public use case. The previous one improved performance for the MP's home page (for logged-in users).

@Bdegraaf1234 I think the SEP meeting is cancelled today as a few of us are on annual leave.

@this-Aditya Please take a look at some comments below.

File: src/main/java/org/radarbase/management/web/rest/PublicResource.kt [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    @GetMapping("projects")
+    @Timed
+    fun getProjectsInfo(
+        @PageableDefault(size = Int.MAX_VALUE) pageable: Pageable,
+    ): ResponseEntity<*> {
+        log.debug("REST request to get Projects for public endpoint")
+        val page = projectService.getPublicProjects(pageable)
+        val headers = PaginationUtil
+            .generatePaginationHttpHeaders(page, "/api/lite-projects")
```
Comment (Bdegraaf1234) [inline src/main/java/org/radarbase/management/web/rest/PublicResource.kt:36]: Is this right?

File: src/main/java/org/radarbase/management/service/dto/PublicProjectDTO.kt [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+class PublicProjectDTO : Serializable {
+    @NonNull
+    var projectName: String? = null
+
+    @NonNull
+    var description: String? = null
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    var sourceTypes: Set<SourceTypeDTO> = emptySet()
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/management/service/dto/PublicProjectDTO.kt:18]: I think here, MinimalSourceTypeDTO should be enough

File: src/main/java/org/radarbase/management/web/rest/PublicResource.kt (hunk 7 above)
Comment (this-Aditya) [reply src/main/java/org/radarbase/management/web/rest/PublicResource.kt]: Ah, sorry for the mistake. I've corrected it in the latest commit.

**Review by yatharthranjan — APPROVED** [review]
Thanks for the updates. Nice work!

LGTM

@mpgxvii do you wish to review?

**Comment by yatharthranjan** [comment]
Merging for now. We can create separate PRs if changes are needed

**Pull request merged**